test/1.4/lib/T_transaction.py

```diff
@@ -35,3 +35,27 @@
                    (Sim_PE_No_Exception, 5, 0x332211, 3))
 stest.expect_equal(b.iface.transaction.issue(trans, 13),
                    Sim_PE_No_Exception)
+
+trans = transaction_t(write=True, inquiry=True, data=bytes(range(20)))
+stest.expect_equal((b.iface.transaction.issue(trans, 17),
+                    b.set_offset, b.set_value, b.set_size),
+                   (Sim_PE_No_Exception, 32, 0x131211100f, 5))
+
+trans = transaction_t(write=True, data=bytes(range(20)))
+stest.expect_equal((b.iface.transaction.issue(trans, 17),
+                    b.write_offset, b.write_value),
+                   (Sim_PE_No_Exception, 32, 0x131211100f))
+
+trans = transaction_t(read=True, inquiry=True, size=20)
+stest.expect_equal((b.iface.transaction.issue(trans, 17),
+                    b.get_offset, b.get_size),
+                   (Sim_PE_No_Exception, 32, 5))
+for x, y in [(0, 7), (7, 15), (15, 20)]:
+    stest.expect_equal(int.from_bytes(trans.data[x:y], 'little'), 4711)
+
+trans = transaction_t(read=True, size=20)
+stest.expect_equal((b.iface.transaction.issue(trans, 17), b.read_offset),
+                   (Sim_PE_No_Exception, 32))
+for x, y in [(0, 7), (7, 15), (15, 20)]:
+    stest.expect_equal(int.from_bytes(trans.data[x:y], 'little'), 4711)
+
```

Comment:
This doesn't test that the splits happen at the right offsets (or does it?)

Reply:
Indeed, it tests that the access is split in correct sizes, and that the last access is on the correct place.
Since the test-device is quite simple the other accesses _could_ be on any random offset.